core/autoscaling.go

```diff
@@ -574,34 +574,28 @@ func (a *autoScalingGroup) bidForSpotInstance(
 		return err
 	}
 
+	// Fetching all required tags prior to launching the SpotRequest
 	spotRequest := resp.SpotInstanceRequests[0]
-	sr := spotInstanceRequest{SpotInstanceRequest: spotRequest,
-		region: a.region,
-		asg:    a,
+	spotRequest.Tags = a.propagatedInstanceTags()
+	spotRequest.Tags = append(spotRequest.Tags, &ec2.Tag{
+		Key:   aws.String("launched-for-asg"),
+		Value: aws.String(a.name),
+	})
+	sr := spotInstanceRequest{
+		SpotInstanceRequest: spotRequest,
+		region:              a.region,
+		asg:                 a,
 	}
```

Comment:
The problem is the tags set on the spotRequest.Tags are never persisted and just remain in memory, as far as I can tell that variable is set just for reading the tags, it can't be directly used for writing them back.

Because of this most of this PR is not going to work, so I'm afraid we need to hold it until the API supports tagging the SIR on creation.